autograder/autograder/scheduler.py

```diff
@@ -0,0 +1,172 @@
+"""The shipper scheduler."""
+
+import json
+import multiprocessing
+import os
+import random
+import shutil
+
+from abc import ABC, abstractmethod
+from typing import List
+
+from .config import Config
+
+
+class Job:
+    """A representation of an autograding job.
+
+    Contains the path to the queue file for this job, as well as a lazily-loaded instance of the
+    contents of its queue file.
+
+    Parameters
+    ----------
+    path : str
+        Path to this job's queue file.
+    """
+    def __init__(self, path: str):
+        self.path = path
+        self._queue_obj = None
+
+    @property
+    def queue_obj(self) -> dict:
+        if self._queue_obj is None:
+            with open(self.path) as f:
+                self._queue_obj = json.load(f)
```

Comment:
Should we handle errors here, or expect calls to this to be wrapped in a try/except?

Reply:
Fixed